runtime/cudaq/platform/default/rest_server/helpers/JsonConvert.h

```diff
@@ -79,9 +208,27 @@ inline void to_json(json &j, const ExecutionContext &context) {
   if (context.expectationValue.has_value()) {
     j["expectationValue"] = context.expectationValue.value();
   }
+
   j["simulationData"] = json();
-  j["simulationData"]["dim"] = std::get<0>(context.simulationData);
-  j["simulationData"]["data"] = std::get<1>(context.simulationData);
+  if (context.simulationState) {
+    j["simulationData"]["dim"] = context.simulationState->getDataShape();
+    std::vector<std::complex<double>> hostData(
+        context.simulationState->getNumElements());
+    if (context.simulationState->isDeviceData()) {
+      context.simulationState->toHost(hostData.data());
+      j["simulationData"]["data"] = hostData;
+    } else {
+      auto *ptr = reinterpret_cast<std::complex<double> *>(
+          context.simulationState->ptr());
+      j["simulationData"]["data"] = std::vector<std::complex<double>>(
+          ptr, ptr + context.simulationState->getNumElements());
```

Comment:
How big will this JSON approach scale? Will it convert everything to ASCII?

Reply:
This is an issue that pre-existed this PR (even worse with more copies). `get_state` for anything other than small toy models with the remote rest qpu does not make much sense.